### PR TITLE
update headless label explanation

### LINKED ISSUE #44440: Confusing explanation for label `service.kubernetes.io/headless `
**This is a Bug Report**

**Problem:** 

> ```markdown
> ### service.kubernetes.io/headless {#servicekubernetesioheadless}
> 
> Type: Label
> 
> Example: `service.kubernetes.io/headless: ""`
> 
> Used on: Service
> 
> The control plane adds this label to an Endpoints object when the owning Service is headless.
> ```

Does this get set on Services, or on EndpointSlices and Endpoints? The docs are not self-consistent.

**Proposed Solution:**

Check the true story and then revise the explanation for this label

**Page to Update:**
https://kubernetes.io/docs/reference/labels-annotations-taints/#servicekubernetesioheadless

**Additional Information:**
/language en
/sig network

### PR DESCRIPTION
**Pull request by tamilselvan1102**
issue #44440

### CODE REVIEW

**Comment by tamilselvan1102** [comment]
preview page: https://deploy-preview-44470--kubernetes-io-main-staging.netlify.app/docs/reference/labels-annotations-taints/#servicekubernetesioheadless

**Comment by Gauravpadam** [comment]
Thanks for the PR @tamilselvan1102

I see that the commit message contains a closing term (resolve) and the issue tag, could you change it otherwise?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -1005,6 +1005,14 @@ Used on: Service
 
 The control plane adds this label to an Endpoints object when the owning Service is headless.
 
+A headless service does not have a cluster IP assigned to it. Instead of providing a single 
+virtual IP address for the service, a headless service creates a DNS record for each pod 
+associated with the service and you can access each pod associated with the service through DNS. 
+
+To create a headless service, we need to define a service with the clusterIP field set to "None".
```
Comment (Gauravpadam) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1012]:
```suggestion
To create a headless service, define a service with the clusterIP field set to **None**. 
```

**Review by Gauravpadam — COMMENTED** [review]
Thanks

Just a few suggestions

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -1005,6 +1005,14 @@ Used on: Service
 
 The control plane adds this label to an Endpoints object when the owning Service is headless.
 
+A headless service does not have a cluster IP assigned to it. Instead of providing a single 
+virtual IP address for the service, a headless service creates a DNS record for each pod 
+associated with the service and you can access each pod associated with the service through DNS. 
+
+To create a headless service, define a service with the clusterIP field set to **None**. 
+Once the headless service has been created, The DNS record for each pod will be in the format 
+`<pod-name>.<headless-service-name>.<namespace>.svc.cluster.local`
```
Comment (stmcginnis) [inline content/en/docs/reference/labels-annotations-taints/_index.md:1014]: nit: should probably add a period to the end of the sentence.

**Review by stmcginnis — COMMENTED** [review]
/lgtm

**Review by kumarankit999 — COMMENTED** [review]
Looks Fixed now!!

**Comment by nate-double-u** [comment]
The commit error message is referring to the number in the commit--One way to fix would be to squash the commits in this PR, and remove the reference to the issue while you're doing it.

**Review by kumarankit999 — COMMENTED** [review]
How much is it necessary to update the word from Uppercase to lowercase on lines 119 & 482?
Also, I see some Conflicts to merge this PR!

**Comment by tamilselvan1102** [comment]
Sorry Squash went wrong. so closing this PR and opened new.
Please review https://github.com/kubernetes/website/pull/45035
Sorry for the inconvenient

**Pull request closed without merging**